caldp/process.py

```diff
@@ -709,17 +876,156 @@ def raw_files(self, files):
 
 # ............................................................................
 
-MANAGERS = {"acs": AcsManager, "cos": CosManager, "stis": StisManager, "wfc3": Wfc3Manager}
+
+class SvmManager(Manager):
+    """
+    Attributes
+    ----------
+    runsinglehap : str
+        (class) Program name to run SVM workflow on dataset_input.out file.
+
+    dataset  : str
+        (instance) Name of dataset being processed
+    output_uri : str
+        (instance) Root output path,  e.g. s3://bucket/subdir/subdir/.../subdir
+    input_uri : str
+        (instance) root input path, or astroquery:// to download from MAST
+
+
+    Methods
+    -------
+    __init__(dataset, input_uri, output_uri)
+
+    download()
+        Downloads data files for `ipppssoot` from astroquery (not implemented)
+    process_inputs(input_files)
+        Runs SVM workflow on dataset_input.out file.
+    """
+
+    def __init__(self, dataset, input_uri, output_uri):
+        super().__init__(dataset, input_uri, output_uri)
+        self.ipppss = get_svm_obs_set(self.dataset)
+        self.runsinglehap = "runsinglehap"
+        self.create_file_search_patterns()
+
+    def create_file_search_patterns(self):
+        self.input_search_patterns = [f"{self.dataset.lower()}_input.out"]
+        self.output_search_patterns = [
+            f"hst_*{self.ipppss.lower()}*.fits",
+            f"hst_*{self.ipppss.lower()}*.txt",
+            f"hst_*{self.ipppss.lower()}*.ecsv",
+            f"{self.dataset.lower()}_manifest.txt",
+            "astrodrizzle.log",
+        ]
+        self.search_input_tar_pattern = f"{self.dataset.lower()}*.tar.gz"
+        self.s3_tar_key = self.dataset.lower() + ".tar.gz"
+
+    # -----------------------------------------------------------------------------
+
+    def process_inputs(self, input_files):
+        """Run runsinglehap on dataset_input.out"""
+        input_file = input_files[0]  # get_input_files returns a list
+        return self.run(exit_codes.SVM_ERROR, self.runsinglehap, input_file)
+
+    # -----------------------------------------------------------------------------
+
+    def download(self):
+        # Not Implemented
+        """Called if input_uri starts is `astroquery`
+        Download any data files for the `dataset`,  issuing start and
+        stop divider messages.
+
+        Returns
+        -------
+        filepaths : sorted list
+            Local file system paths of files which were downloaded for `dataset`.
+        """
+        pass
```

Comment:
Question:  is it possible to astroquery for the new input data?   If so,  that's potentially very useful for CI testing in environments like GitHub.    Either way,  it may be better to  raise NotImplementedError than pass.

Reply:
This might be a bit tricky because we may need the _input.out file to tell us which files to download for each dataset. This is related to the question @bhayden53 asked, and I agree it's a good thing to implement if possible. I can look into this further. For now I will change this to raise NotImplementedError.

Reply:
We should probably try to identify some small HAP datasets that can be used for testing, and see what we can do. But perhaps that's getting too big for this one PR and needs it's own ticket.

Reply:
No objections here.   I do think we could benefit from tickets which snapshot Jenny's expertise in this eventually,   assuming nobody (particularly me) wrote them:
1. Add unit test coverage for the new code as Brian suggested.
2. Capture dev/test data inputs and/or outputs and upload them to the sandbox pristine inputs bucket.
3. Write a little documentation explaining the new modes, inputs,  and outputs.